### DISCUSSION TITLE
Setting MCO to HSE on STM32F4

### DISCUSSION THREAD

**Discussion by exelsior87** [zephyrproject-rtos/zephyr#80465] [Q&A] [answered]
I'm developing on STM32F429.
I am trying to configure an MCO.

`
&mco1 {
	status = "okay";
	clocks = <&rcc STM32_SRC_HSE MCO1_SEL(MCO1_SEL_HSE)>;
}
`

If I configure it as above, get a syntax error because STM32_SRC_HSE is not defined.
It's defined in the other series,
https://github.com/zephyrproject-rtos/zephyr/blob/5969a10c3064e0e2954c5b5593d6501818618ac0/include/zephyr/dt-bindings/clock/stm32f7_clock.h#L30-L33

but why is STM32_SRC_HSE undefined on STM32F4?
LSE and LSI are defined in the stm32_common_clock.h file, so I can use them in f4.
but HSE is not.

https://github.com/zephyrproject-rtos/zephyr/blob/5969a10c3064e0e2954c5b5593d6501818618ac0/include/zephyr/dt-bindings/clock/stm32_common_clocks.h#L11-L13

What is the proper way to set MCO to HSE on stm32f4?
Do I need to set the register myself?

**Comment by FRASTM** [comment, marked as answer]
@exelsior87 , you're right
Please look at https://github.com/zephyrproject-rtos/zephyr/pull/80509

**Reply by exelsior87** [reply]
thanks
